Panic in CAPH controller

/kind bug

### What steps did you take and what happened:
Application panic in normal use. 

I was repeatedly testing a script to deploy a cluster and discovered that the `Cluster.cluster.x-k8s.io/v1beta1` failed to delete.  Removing the `finalizers` allowed the record to be deleted and the infra provider resumed normal operation. Not sure why the cluster record was unable to be deleted in the first place.

### What did you expect to happen:
The application should not panic

### Anything else you would like to add:

The controller runs in a restart loop consistently showing the following error message.

<details>
<summary>Logs</summary>

```
{"level":"INFO","time":"2024-07-18T09:46:00.106Z","file":"controller/controller.go:186","message":"Starting Controller","controller":"certificatesigningrequest","controllerGroup":"certificates.k8s.io","controllerKind":"CertificateSigningRequest"}
{"level":"INFO","time":"2024-07-18T09:46:00.134Z","file":"controller/controller.go:115","message":"Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference","controller":"hetznercluster","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"HetznerCluster","HetznerCluster":{"name":"beta","namespace":"beta"},"namespace":"beta","name":"beta","reconcileID":"04d90b63-ae3f-4e6a-8880-b141be30d935"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x183d19d]

goroutine 397 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	sigs.k8s.io/controller-runtime@v0.17.3/pkg/internal/controller/controller.go:116 +0x1e5
panic({0x1a71140?, 0x2e98890?})
	runtime/panic.go:914 +0x21f
github.com/syself/cluster-api-provider-hetzner/pkg/services/hcloud/loadbalancer.(*Service).Delete(0xc00099d800, {0x1fe4180, 0xc000df99e0})
	github.com/syself/cluster-api-provider-hetzner/pkg/services/hcloud/loadbalancer/loadbalancer.go:351 +0x7d
github.com/syself/cluster-api-provider-hetzner/controllers.(*HetznerClusterReconciler).reconcileDelete(0xc00021c900, {0x1fe4180, 0xc000df99e0}, 0xc0002ef5e0)
	github.com/syself/cluster-api-provider-hetzner/controllers/hetznercluster_controller.go:337 +0x2c6
github.com/syself/cluster-api-provider-hetzner/controllers.(*HetznerClusterReconciler).Reconcile(0xc00021c900, {0x1fe4180, 0xc000df9770}, {{{0xc000604240?, 0x5?}, {0xc00060423c?, 0xc00099dd08?}}})
	github.com/syself/cluster-api-provider-hetzner/controllers/hetznercluster_controller.go:166 +0xa56
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x1fe75b0?, {0x1fe4180?, 0xc000df9770?}, {{{0xc000604240?, 0xb?}, {0xc00060423c?, 0x0?}}})
	sigs.k8s.io/controller-runtime@v0.17.3/pkg/internal/controller/controller.go:119 +0xb7
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0002f2780, {0x1fe41b8, 0xc00012d0e0}, {0x1b41740?, 0xc000278da0?})
	sigs.k8s.io/controller-runtime@v0.17.3/pkg/internal/controller/controller.go:316 +0x3cc
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0002f2780, {0x1fe41b8, 0xc00012d0e0})
	sigs.k8s.io/controller-runtime@v0.17.3/pkg/internal/controller/controller.go:266 +0x1af
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	sigs.k8s.io/controller-runtime@v0.17.3/pkg/internal/controller/controller.go:227 +0x79
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2 in goroutine 168
	sigs.k8s.io/controller-runtime@v0.17.3/pkg/internal/controller/controller.go:223 +0x565
```

</details>


### Environment:

- cluster-api-provider-hetzner version: v1.0.0-beta.37 (tried beta.36 as well)
- Kubernetes version: v1.29.2
- OS (e.g. from `/etc/os-release`): ubuntu jammy 22.04.4 LTS

--- a/pkg/services/hcloud/loadbalancer/loadbalancer.go
+++ b/pkg/services/hcloud/loadbalancer/loadbalancer.go
@@ -348,24 +348,15 @@ func (s *Service) Delete(ctx context.Context) (err error) {
 
 	// do not delete a protected load balancer or one that has not been created by this controller
 	if s.scope.HetznerCluster.Status.ControlPlaneLoadBalancer.Protected || s.scope.HetznerCluster.Spec.ControlPlaneLoadBalancer.Name != nil {
-		name := *s.scope.HetznerCluster.Spec.ControlPlaneLoadBalancer.Name
-
-		loadBalancers, err := s.scope.HCloudClient.ListLoadBalancers(ctx, hcloud.LoadBalancerListOpts{Name: name})
+		lb, err := s.findLoadBalancer(ctx)
 		if err != nil {
-			hcloudutil.HandleRateLimitExceeded(s.scope.HetznerCluster, err, "ListLoadBalancers")
-			return fmt.Errorf("failed to list load balancers: %w", err)
-		}
-
-		if len(loadBalancers) > 1 {
-			return fmt.Errorf("found %v load balancers in HCloud with name %q", len(loadBalancers), name)
+			return fmt.Errorf("failed to find load balancer: %w", err)
 		}
 
 		// nothing to do if load balancer is not found
-		if len(loadBalancers) == 0 {
+		if lb == nil {
 			return nil
 		}
-
-		lb := loadBalancers[0]
 
 		// remove owned label and update
 		delete(lb.Labels, s.scope.HetznerCluster.ClusterTagKey())

